pipelines/short-read-alignment-vector/farm5/ShortReadAlignment.wdl

```diff
@@ -28,37 +33,57 @@ workflow ShortReadAlignment {
     RunTimeSettings runTimeSettings
   }
 
-  scatter (idx in range(length(input_fastq1s))) {
-    call Tasks.ReadAlignment {
-      input:
-        sample_id = sample_id,
-        read_group_id = read_group_ids[idx],
-        fastq1 = input_fastq1s[idx],
-        fastq2 = input_fastq2s[idx],
-        output_sam_basename = output_basename,
-        reference = reference,
-        runTimeSettings = runTimeSettings
+  # Note that the order of input_crams, input_bams, and input_fastqs are all optional.
+  # If more than one of them is provided, the order of precedence is:
+  # crams first (if found)
+  # bams next (if found)
+  # fastqs last
+  Array[File] input_files = select_first([input_crams, input_bams, input_fastq1s])
+
+  scatter (idx in range(length(input_files))) {
+    if (defined(input_crams)) {
```

Comment:
Do we need to differentiate here? It looks like you could call `Alignment.Alignment` and just pass all the optional files  and the `output_basename` through (since Alignment also checks for the crams, bams, and fastqs). This does have the advantage of making it more clear which files will be used if multiple are provided

Reply:
Yeah, that might be clearer.  I did it this way as it made the logic simpler in Alignment.  If I change Alignment to do all the differentiation it needs to do things like
`if (defined(input_cram) && (!defined(input_bam) && !defined(input_fastq)) 
`
which is not as straightforward as that, since wdl doesn't have an else.

Reply:
Since we are preferentially selecting the cram, then bam, then fastq formats, I believe the alignment wdl as it is currently written will do the correct thing. On the other hand, this method will make it clear in the metadata which input files were actually used. I think either way is fine

Reply:
So, actually the problem I was running into is that the conditional here is based on if (defined(input_crams)), if I tried to call Alignment with
select_first([input_crams])[idx], select_first([input_bams])[idx], ...
it would get an error if either one of those wasn't defined.

Reply:
Makes sense